pandas/core/arrays/datetimes.py

```diff
@@ -222,6 +222,12 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
     @classmethod
     def _generate_range(cls, start, end, periods, freq, tz=None,
                         normalize=False, ambiguous='raise', closed=None):
+
+        periods = dtl.validate_periods(periods)
+        if freq is None and any(x is None for x in [periods, start, end]):
+            raise ValueError('Must provide freq argument if no data is '
+                             'supplied')
+
```

Comment:
Moved these checks here from inside `__new__`

Reply:
+1, as long as we're OK with ignoring gibberish arguments when there not actually used (which seems to be what we do right now).

```python

In [9]: idx = pd.PeriodIndex(['2000', '2001'], freq='D')

In [10]: pd.PeriodIndex(idx, start='foo')
Out[10]: PeriodIndex(['2000-01-01', '2001-01-01'], dtype='period[D]', freq='D')
```